From 6c53c614d9522f20a09236b0901f4848d2bc17a4 Mon Sep 17 00:00:00 2001
Message-Id: <6c53c614d9522f20a09236b0901f4848d2bc17a4.1421886431.git.chang-joon.lee@intel.com>
In-Reply-To: <2a59efb195329ac28200fa6e68e1170a4cf8d03c.1421886431.git.chang-joon.lee@intel.com>
References: <2a59efb195329ac28200fa6e68e1170a4cf8d03c.1421886431.git.chang-joon.lee@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Wed, 10 Dec 2014 06:18:04 +0530
Subject: [PATCH 02/11] SQUASHME! [VPG]: drm/i915: Removed the vbt_version
 flag

SQUASHME! - This patch should be squashed into the following exising patch:
	Author: Deepak M <m.deepak@intel.com>
	Change-Id: Ib6594e7de9b0c7ad2ee490f4c3c934c176a420e1
	FOR_UPSTREAM [VPG]: drm/i915: Parsing LPF brightness control from VBT

VBT version can be read from the bdb_header structure
so removing the vbt_version variable.

Issue: IRDA-2088
Change-Id: Ic61eda568c2a69648942443e9daa7b82bb6d0eef
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h   |    1 -
 drivers/gpu/drm/i915/intel_bios.c |    5 +----
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index c0cafcb..9f3ee5a 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1622,7 +1622,6 @@ struct drm_i915_private {
 	struct i915_drrs drrs;
 	struct intel_opregion opregion;
 	struct intel_vbt_data vbt;
-	u16 vbt_version;
 	bool scaling_reqd;
 
 	struct kobject memtrack_kobj;
diff --git a/drivers/gpu/drm/i915/intel_bios.c b/drivers/gpu/drm/i915/intel_bios.c
index 0025a04..d329e4e 100644
--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -352,7 +352,7 @@ parse_lfp_backlight(struct drm_i915_private *dev_priv, struct bdb_header *bdb)
 
 	dev_priv->vbt.backlight.pwm_freq_hz = entry->pwm_freq_hz;
 	dev_priv->vbt.backlight.active_low_pwm = entry->active_low_pwm;
-	if (dev_priv->vbt_version < 191)
+	if (bdb->version < 191)
 		dev_priv->vbt.backlight.controller = DISPLAY_DDI_BKL_CNTL;
 	else {
 		dev_priv->vbt.backlight.pin = brightness->pin;
@@ -1298,7 +1298,6 @@ intel_parse_bios(struct drm_device *dev)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct pci_dev *pdev = dev->pdev;
 	struct bdb_header *bdb = NULL;
-	struct vbt_header *vbt;
 	u8 __iomem *bios = NULL;
 
 	if (HAS_PCH_NOP(dev))
@@ -1312,8 +1311,6 @@ intel_parse_bios(struct drm_device *dev)
 		bdb = validate_vbt((char *)dev_priv->opregion.header, OPREGION_SIZE,
 				   (struct vbt_header *)dev_priv->opregion.vbt,
 				   "OpRegion");
-		vbt = (struct vbt_header *)dev_priv->opregion.vbt;
-		dev_priv->vbt_version = vbt->version;
 	}
 
 	if (bdb == NULL) {
-- 
1.7.9.5

